cylc/flow/data_store_mgr.py

```diff
@@ -1126,7 +1196,7 @@ def get_publish_deltas(self):
         result.append(
             (ALL_DELTAS.encode('utf-8'), all_deltas, 'SerializeToString')
         )
-        return result
+        return deepcopy(result)
```

Comment:
Why this deepcopy?

Reply:
So the data elements aren't modified between gathering them here and publishing them.

Reply:
Fair enough.

Reply:
I'll have to check if there's even a chance that they could be.. Might not be needed, probably safer, might depend on whether they need accessed aysnc to other things and how ZMQ publisher works..

Reply:
(That's one of the things I'd like to understand better).